server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -413,6 +470,46 @@ private CompletionStage<Result> renderErrorOrRedirectToNextBlock(
                           ApplicantQuestionRendererParams.ErrorDisplayMode.DISPLAY_ERRORS))));
     }
 
+    if (featureFlags.isEsriAddressCorrectionEnabled(request)
+        && thisBlockUpdated.hasAddressWithCorrectionEnabled()) {
+
+      ApplicantQuestion applicantQuestion =
+          applicantService.getFirstAddressCorrectionEnabledApplicantQuestion(thisBlockUpdated);
+
+      AddressQuestion addressQuestion = applicantQuestion.createAddressQuestion();
+
+      if (addressQuestion.needsAddressCorrection(applicantQuestion.isAddressCorrectionEnabled())) {
+
+        return applicantService
+            .getAddressSuggestionGroup(thisBlockUpdated)
+            .thenApplyAsync(
+                addressSuggestionGroup -> {
+                  ImmutableList<AddressSuggestion> suggestions =
+                      addressSuggestionGroup.getAddressSuggestions();
+                  String json = addressSuggestionJsonSerializer.serialize(suggestions);
+
+                  Boolean isEligibilityEnabled =
+                      thisBlockUpdated.getLeafAddressNodeServiceAreaIds().isPresent();
```

Comment:
**suggestion (blocking):** use the feature flags class to check if a feature is enabled or not

Reply:
Oh this isn't checking if the feature is enabled. It's checking if this block is configured with an eligibility check. I renamed the variable to `isEligibilityEnabledOnThisBlock`.

Reply:
ah sounds good